addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -454,52 +453,67 @@ var FileWidget = SearchableMediaWidget.extend({
+                src += _.str.sprintf('?access_token=%s', img.access_token);
             }
+            if (!this.$media.is('img')) {
 
-            self.$media.attr('alt', img.alt);
-            var style = self.style;
-            if (style) {
-                self.$media.css(style);
+                // Note: by default the images receive the bootstrap opt-in
+                // img-fluid class. We cannot make them all responsive
+                // by design because of libraries and client databases img.
+                this.$media = $('<img/>', {class: 'img-fluid o_we_custom_image'});
+                this.media = this.$media[0];
             }
-
-            // Remove crop related attributes
-            if (self.$media.attr('data-aspect-ratio')) {
-                var attrs = ['aspect-ratio', 'x', 'y', 'width', 'height', 'rotate', 'scale-x', 'scale-y', 'crop:originalSrc'];
-                self.$media.removeClass('o_cropped_img_to_save');
-                _.each(attrs, function (attr) {
-                    self.$media.removeData(attr);
-                    self.$media.removeAttr('data-' + attr);
-                });
+            this.$media.attr('src', src);
+        } else {
+            if (!this.$media.is('a')) {
+                $('.note-control-selection').hide();
+                this.$media = $('<a/>');
+                this.media = this.$media[0];
             }
-            return self.media;
+            var href = '/web/content/' + img.id + '?';
+            if (!img.public && img.access_token) {
+                href += _.str.sprintf('access_token=%s&', img.access_token);
+            }
+            href += 'unique=' + img.checksum + '&download=true';
+            this.$media.attr('href', href);
+            this.$media.addClass('o_image').attr('title', img.name).attr('data-mimetype', img.mimetype);
+        }
+
+        this.$media.attr('alt', img.alt);
+        var style = this.style;
+        if (style) {
+            this.$media.css(style);
+        }
+
+        // Remove crop related attributes
+        if (this.$media.attr('data-aspect-ratio')) {
+            var attrs = ['aspect-ratio', 'x', 'y', 'width', 'height', 'rotate', 'scale-x', 'scale-y', 'crop:originalSrc'];
+            this.$media.removeClass('o_cropped_img_to_save');
+            attrs.forEach(attr => {
+                this.$media.removeData(attr);
+                this.$media.removeAttr('data-' + attr);
+            });
+        }
+        return this.media;
+    },
+    /**
+     * Creates and returns an optimized copy of an attachment.
+     *
+     * @private
+     * @param {object} attachment
+     */
+    _optimizeAttachment: function (attachment) {
+        return this._rpc({
+            route: `/web_editor/attachment/${attachment.id}/update`,
+            params: {
+                copy: true,
+                name: attachment.name + this.OPTIMIZED_SUFFIX,
+                quality: 80,
+                width: this._computeOptimizedWidth(),
```

Comment:
I am not sure, did we fix that optimizedWidth suggestion in 13.0?

Reply:
No, unfortunately. It still just does `min(options.mediaWidth, 1920)`, which is only ever set to 1920 anyway. When it's not set, the method returns NaN, which is then ignored for resizing. I had some code in the old branch to somewhat deal with that (the thing with bootstrap columns and considering whether we were in a container and all that), which I had planned to add in part 2, but maybe we need to fix this in stable? Not sure.

Reply:
Yes, we need to fix in standard. I guess we lost the use of `mediaWidth` with the editor revert at some point, as it was working before. If you have time, you can have a look :+1: